car_control:
  ros_parameters:
    speed_setpoint: 1.0       # m/s
    max_speed: 2.0            # m/s
    heartbeat_timeout: 0.5    # seconds
    max_steer: 0.65           # radians

    wheelbase: .404           # m

    control_method: "youla" # controller type SELECT FROM: stanley | youla | mpc | hinf | pp

    stanley_k: 1              # stanley controller gain
    stanley_k_soft: 1         # stanley softening gain

    mpc_tf: 1.0               # time horizon [sec]
    mpc_N: 20                 # number of shooting nodes                
    mpc_nx: 7                 # number of states
    mpc_nu: 1                 # number of inputs

    pp_lookahead: 1.0         # pure pursuit lookahead dist [m]

    Youla_Gca: [1.0 0.0 0.0 0.0; 0.0 1.0 0.0 0.0; 0.0 0.0 1.0 0.0; 0.0 0.0 0.0 1.0]
    Youla_GcB: [1.0; 0.0; 0.0; 0.0]
    Youla_GcC: [1.0 0.0 0.0 0.0]
    Youla_GcD: [0.0]

    hinf_matrices: "Hinf_Controller.json"   # json title of Hinf Matrices

    waypoints: "waypoints.json"             # json title of waypoints

    

